### PR TITLE
Unify OnCall and Webhook v1 configs

### PR DESCRIPTION
**Pull request by yuri-tceretian**
## Summary
- Replace duplicated raw settings structs with tagged Config types and JSON/YAML decoding adapters.
- Preserve maxAlerts conversion, payload pointer semantics, defaults, decryption, validation order, and partial error results.
- Preserve webhook TLS ServerName behavior and restricted-header checks.
- Check schemas against Config while retaining recursive coverage of the separately parsed HTTP client settings.
- Add compatibility tests for numeric edge cases, YAML scalars/aliases, payload null/merge behavior, and TLS/HMAC decryption.

## Testing
- `make lint`
- `make test`
- Reviewer agent re-review: no remaining actionable findings.

Related to grafana/alerting-squad#1214.
Stacked on #687.

### CODE REVIEW

File: notify/schema_invariants_test.go [hunk 1]
```diff
@@ -68,16 +68,30 @@ func TestIntegrationSchemasMatchConfigStructs(t *testing.T) {
...
-						var httpConfig map[string]any
-						require.NoError(t, json.Unmarshal([]byte(notifytest.FullValidHTTPConfigForTesting), &httpConfig))
-						config["http_config"] = httpConfig["http_config"]
-					}
 					compareFieldsToJSON(t, "", version.Options, []map[string]any{config})
 					return
 				}
-				compareFieldsToStruct(t, "", version.Options, configType)
+				options := version.Options
+				if integration.Type == schema.WebhookType && version.Version == schema.V1 {
```
Comment (yuri-tceretian) [inline notify/schema_invariants_test.go:75]: This will be removed later
